### DISCUSSION TITLE
Error on homepage link

### DISCUSSION THREAD

**Discussion by levipadre** [vercel/next.js#50676] [Help] [answered]
### Summary

When I click on the homepage nextjs link: `<Link href='/'>Home</Link>` the url doesn't change but the page has gone, and in console log it gives me the following error: `An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.`.

The problem only occurs in production.

**Comment by devjiwonchoi** [comment]
Are you able to check the `logs` where you have hosted?

Also, does it throw any error on the local production build by `next build` and then `next start`?

**Reply by levipadre** [reply]
You are right, I don't know why I haven't checked the log yet :)
It is hosted on Vercel. I doesn't throw any error when I use local production, however on live Vecel logs says:
```
- info Loaded env from /var/task/.env
TypeError: Cannot read properties of undefined (reading 'meta')
    at Module.generateMetadata (/var/task/.next/server/app/[[...slug]]/page.js:455:21)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

But when I load the homepage from url, it works.

I have a default generated metadata on layout.tsx and I have generated metadata on each pages `app/[[...slug]]/page.js`, where I only override the title, description and keywords:
```
export async function generateMetadata({
    params
}: {
    params: {
        slug: string[]
    }
}, parent: ResolvingMetadata): Promise<Metadata> {
    const [META] = await Promise.all([getPageByUrl(params.slug)]);

    return {
        title: META.meta.title,
        description: META.meta.description,
        keywords: META.meta.keywords
    };
}
```

**Comment by levipadre** [comment]
I think I found it. Homepage's slug for some reason is [ 'index' ], so when I matched my feed against slug I assumed it's [ '' ] but it wasn't.

**Reply by devjiwonchoi** [reply]
Glad you figured it out :)
Please let me know if that resolves the issue.

**Reply by levipadre** [reply, marked as answer]
If anyone is having the same issue, one production, the homepage's slug is [ '' ] if you load the page from URL or reload the homepage, but if you navigate to it by nextjs Link, it's [ 'index' ]

So in my case, I couldn't fetch the page data by slug because it didn't match.

With this fetch, it worked:
```
const fetcher = async (endpoint: string, params: Object) => {
    const res = await fetch(`https://domain.com/api/${endpoint}`, params);
  
    if (!res.ok) {
        // This will activate the closest `error.js` Error Boundary
        throw new Error('Failed to fetch data');
    }

    return res.json();
};

export async function getPageByUrl(slug: string[]) {
    const path = (typeof slug === 'undefined' || slug[0] === 'index') ? '' : slug.join('/');

    const result = await fetcher(`pages?url=/${path}`, {});
    
    return result[0];
}
```

Maybe the code could be better, but it does the job now. Any suggestion or optimisation would be welcome.
